src/ccache.cpp

```diff
@@ -1805,10 +1815,11 @@ set_up_config(Config& config)
       config.set_cache_dir(default_cache_dir(home_dir));
     }
   }
-  // else: cache_dir was set explicitly via environment or via secondary config.
```

Comment:
Hups I didn't even notice all those formatting changes. The magic of new clang-format versions :-(

Best guess: I'm running a linux clang-format on a windows file system with '\r\n' file endings, so it counts '\r' as an extra char. Will try to reproduce tomorrow.
It passes CI, so it can't be that bad :-)

Reply:
Ok it took me a while to reproduce this. It's not a bug. Technically.
It's kind of due to the automatic format on save. Not everything one saves is correct. But the compiler tells that only later. By then the file has been reformatted. There is no "reformat automatically, but only if it compiles" option.

Exact scenario: remove one `}` on top of a file. Save it. It will be reformatted. Obviously it will not compile. Add the missing `}`. File is reformatted again. But the comment reformatting stays, since clang-format does not reformat too short comments.

So the options are:
* Do not format on save automatically
* Review what you push. Obviously that's a good idea anyway... 😉

Reply:
> Exact scenario: remove one `}` on top of a file. Save it. It will be reformatted. Obviously it will not compile. Add the missing `}`. File is reformatted again. But the comment reformatting stays, since clang-format does not reformat too short comments.

Ah! :+1: When I do that mistake I've learned to use undo instead of just fixing the curly braces, but it indeed took some time to get into that habit. :slightly_smiling_face: